src/main/drivers/transponder_ir.h

```diff
@@ -17,29 +17,70 @@
 
 #pragma once
 
+#define BIT_TOGGLE_1 78
+#define BIT_TOGGLE_0 0
+
+
+/*** ARCITIMER ***/
+#define TRANSPONDER_BITS_PER_BYTE_ARCITIMER 8
+#define TRANSPONDER_DATA_LENGTH_ARCITIMER 9
+#define TRANSPONDER_TOGGLES_PER_BIT_ARCITIMER 4
+#define TRANSPONDER_GAP_TOGGLES_ARCITIMER 0
+#define TRANSPONDER_TOGGLES_ARCITIMER (TRANSPONDER_TOGGLES_PER_BIT_ARCITIMER + TRANSPONDER_GAP_TOGGLES_ARCITIMER)
+
+#define TRANSPONDER_DMA_BUFFER_SIZE_ARCITIMER 155 * TRANSPONDER_TOGGLES_PER_BIT_ARCITIMER // 620
+/*** ******** ***/
+
+/*** ILAP ***/
 #define TRANSPONDER_BITS_PER_BYTE 10 // start + 8 data + stop
 #define TRANSPONDER_DATA_LENGTH 6
 #define TRANSPONDER_TOGGLES_PER_BIT 11
 #define TRANSPONDER_GAP_TOGGLES 1
 #define TRANSPONDER_TOGGLES (TRANSPONDER_TOGGLES_PER_BIT + TRANSPONDER_GAP_TOGGLES)
 
-#define TRANSPONDER_DMA_BUFFER_SIZE ((TRANSPONDER_TOGGLES_PER_BIT + 1) * TRANSPONDER_BITS_PER_BYTE * TRANSPONDER_DATA_LENGTH)
+#define TRANSPONDER_DMA_BUFFER_SIZE ((TRANSPONDER_TOGGLES_PER_BIT + 1) * TRANSPONDER_BITS_PER_BYTE * TRANSPONDER_DATA_LENGTH) //720
+/*** ******** ***/
 
-#define BIT_TOGGLE_1 78 // (156 / 2)
-#define BIT_TOGGLE_0 0
 
-void transponderIrInit(void);
+typedef union transponderIrDMABuffer_s {
+
+    uint8_t arcitimer[TRANSPONDER_TOGGLES_ARCITIMER]; // 620
+    uint8_t ilap[TRANSPONDER_DMA_BUFFER_SIZE]; // 720
+
+} transponderIrDMABuffer_t;
+
+typedef struct transponder_s {
+
+    uint8_t bits_per_byte;
+    uint8_t data_length;
+    uint8_t toggles_per_bit;
+    uint8_t gap_toggles;
+    uint32_t dma_buffer_size;
+    uint8_t TIM_Prescaler;
+    transponderIrDMABuffer_t transponderIrDMABuffer; // private member
+    const struct transponderVTable *vTable;
+
+} transponder_t;
+
+struct transponderVTable {
+
+    void (*updateTransponderDMABuffer)(transponder_t* transponder, const uint8_t* transponderData);
+
+};
+
+void transponderIrInit(const uint8_t* transponderType);
 void transponderIrDisable(void);
 
-void transponderIrHardwareInit(void);
-void transponderIrDMAEnable(void);
+void transponderIrHardwareInit(transponder_t* transponder_l);
+void transponderIrDMAEnable(transponder_t* transponder_l);
```

Comment:
the `_l` in `transponder_l` isn't helpful in the context of a function declaration used to describe the API.  just `transponder` or `instance` is probably better.